django/contrib/admin/sites.py

```diff
@@ -45,7 +45,7 @@ class AdminSite:
     # Text to put at the end of each page's <title>.
     site_title = gettext_lazy("Django site admin")
 
-    # Text to put in each page's <h1>.
+    # Text to put in each page's <div id="site-name">.
```

Comment:
Docs update is also require with `versionchanged:: 5.0` annotation, e.g.
```
diff --git a/docs/ref/contrib/admin/index.txt b/docs/ref/contrib/admin/index.txt
index d3c3884329..f6a61d3eca 100644
--- a/docs/ref/contrib/admin/index.txt
+++ b/docs/ref/contrib/admin/index.txt
@@ -2848,9 +2848,13 @@ Templates can override or extend base admin templates as described in
 
 .. attribute:: AdminSite.site_header
 
-    The text to put at the top of each admin page, as an ``<h1>`` (a string).
+    The text to put at the top of each admin page, as an ``<div>`` (a string).
     By default, this is "Django administration".
 
+    .. versionchanged:: 5.0
+
+        In older versions, ``site_header`` was using a ``<h1>`` tag.
+
 .. attribute:: AdminSite.site_title
 
     The text to put at the end of each admin page's ``<title>`` (a string). By
```
We should also document this as a backward incompatible change in Django 5.0 release notes.